client/client-lib/src/lib.rs

```diff
@@ -458,7 +487,10 @@ impl Client<UserClientConfig> {
         invoice: Invoice,
         mut rng: R,
     ) -> Result<(ContractId, OutPoint)> {
-        let gateway = self.fetch_gateway().await?;
+        let gateway = match self.fetch_active_gateway().await {
+            Some(g) => g,
+            None => self.switch_active_gateway(None).await?,
```

Comment:
if we don't use option anymore you'd need to `fetch_registered_gateways` here and then just give it the pubkey (and if there are no gateways return error)

Reply:
```
   let gateway = match self.fetch_active_gateway().await {
            Some(g) => g,
            None => {
                let gws = self.fetch_registered_gateways().await?;
                if let Some(gw) = gws.first() {
                    self.switch_active_gateway(gw.node_pub_key).await?
                } else {
                    panic!("dont panic but return a error");
                }
            }
        };
```